### ISSUE TITLE
taskq hangs the app in version 0.4.0

### ISSUE THREAD

**Issue by lokenorlinjohannessen** [mvisonneau/gitlab-ci-pipelines-exporter#154]
## Problem
I upgraded gitlab-ci-pipelines-exporter from 0.3.3 to 0.4.0, and now it "hangs" when trying to PullProjectRefsFromProjects

I have a little over 1100 projects, so that might be part of the reason ( But it worked fine in 0.3.3 )

## Error
```go
created by github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter.schedulePullProjectRefsFromProjects
	github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter/scheduler.go:120 +0x398
goroutine 188439 [chan send, 51 minutes]:
github.com/vmihailenco/taskq/v3.(*Consumer).Add(...)
	github.com/vmihailenco/taskq/v3@v3.0.0/consumer.go:144
github.com/vmihailenco/taskq/v3/memqueue.(*Queue).enqueueMessage(0xc0002240c0, 0xc007c2f790, 0xc009662d00, 0xc009662d50)
	github.com/vmihailenco/taskq/v3@v3.0.0/memqueue/queue.go:209 +0x13d
github.com/vmihailenco/taskq/v3/memqueue.(*Queue).Add(0xc0002240c0, 0xc007c2f790, 0xc67d40, 0xc007d70240)
	github.com/vmihailenco/taskq/v3@v3.0.0/memqueue/queue.go:184 +0xdd
github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter.schedulePullProjectRefsFromProject(0xdfc560, 0xc000292000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter/scheduler.go:166 +0x183
created by github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter.schedulePullProjectRefsFromProjects
	github.com/mvisonneau/gitlab-ci-pipelines-exporter/pkg/exporter/scheduler.go:120 +0x398
goroutine 188440 [chan send, 51 minutes]:
```

**Comment by mvisonneau** [comment]
👋 hey @lokenorlinjohannessen, thanks for reporting this issue!

Am I right to assume that the logs you provided are from the profiling endpoint? 🤔 
Does it happen right away or after a while?

**Comment by lokenorlinjohannessen** [comment]
@mvisonneau actually the logs are directly from STDOUT.
It took roughly around 6hours before it started to hang.

Config I used:
```
gitlab:
  url: https://XXX
defaults:
  fetch_pipeline_job_metrics: true
  output_sparse_status_metrics: true
  refs_regexp: '.*'
  fetch_merge_request_pipelines_refs: true
  output_sparse_status_metrics: true
wildcards:
  - owner:
      name: XXX
      kind: group
  - owner:
      name: XXX
      kind: group
  - owner:
      name: XXX
      kind: group
      include_subgroups: true
  - owner:
      name: XXX
      kind: group
  - owner:
      name: XXX
      kind: group
      include_subgroups: true
  - owner:
      name: XXX
      kind: group

```

**Comment by mvisonneau** [comment]
Interesting thanks!

Well, have a look at the configuration syntax as it has changed quite a bit and you will need to refactor what is under `defaults`. (definitely not related to the issue your are facing though)

Regarding the error, do you have metrics around the memory usage of your container?

**Comment by mvisonneau** [comment]
I pushed a change which should disable some system resources check which were being arbitrarily defined. https://github.com/mvisonneau/gitlab-ci-pipelines-exporter/commit/01ca731c6cd0dce6eebd83ff5596bcba6d6272b5#diff-fb4371170436403a2fe1abe0dfe58b4701c74866eb9237670df0621184983f6fR69

If you can manage to try it out with the latest version and confirm if it sort this issue out or not that would be super helpful! 🙇

**Comment by lokenorlinjohannessen** [comment]
> Regarding the error, do you have metrics around the memory usage of your container?

 What metrics to you need, since I got prometheus scraping the endpoint, I have all of the golang metrics stored.

> If you can manage to try it out with the latest version...

Jep, going to try it tomorrow ( I am ending my workday soon ) :D

**Comment by mvisonneau** [comment]
> What metrics to you need, since I got prometheus scraping the endpoint, I have all of the golang metrics stored.

I was looking to check the amount of buffered and free memory, if you have the 1m CPU load & CPU count that would also have been quite interesting.

Have a good evening! :)

**Comment by mvisonneau** [comment]
@lokenorlinjohannessen, I just released [0.4.1](https://github.com/mvisonneau/gitlab-ci-pipelines-exporter/releases/tag/0.4.1), in case it makes it easier for you to try out!

**Comment by mvisonneau** [comment]
@lokenorlinjohannessen, I was never able to reproduce this issue since 0.4.1, closing this issue. Feel free to reopen if necessary.
